lib/memoizable/memory.rb

```diff
@@ -97,7 +97,9 @@ def fetch(name)
     #
     # @api public
     def key?(name)
-      @memory.key?(name)
+      @monitor.synchronize do 
+        @memory.key?(name)
+      end
```

Comment:
__question:__ @mbj do you think this is necessary? This `Memoizable::Memory#key?` method is called in this class within `#[]=` as well as within `Memoizable::ModuleMethods#memoized?`.

All writes are synchronized, but I am wondering what would happen if another thread tries to test for the key presence without synchronization.

Reply:
I think you should remove the entire method, as I cannot think about a legitimate call side scenario. 

Aside from that: It should not be necessary to synchronize the key method as both in GIL and non GIL scenario the answer is meaningless if you ever get `false`.

Reply:
> I think you should remove the entire method, as I cannot think about a legitimate call side scenario.

@mbj it's used In `Memoizable::ModuleMethods#memoized?`. I have tests in other projects that check to see if a method has been memoized (e.g. Adamantium).

> do you think this is necessary?

My question was whether the `@monitor.syncronize` is required to serialize access prior to the key test.

Reply:
> @mbj it's used In `Memoizable::ModuleMethods#memoized?`. I have tests in other projects that check to see if a method has been memoized (e.g. Adamantium).

But this is a boot time API, not a runtime API. I do not see why metaprogramming has to be thread safe?

Meaning: I think this method should not be backed by a `Memory` object at all? Given it does things (synchronization) that are not required? Doing multi threaded metaprogramming is "never" in scope?